src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -16,6 +16,48 @@
 MSI_LOCAL_ID = '[system]'
 
 
+def _validate_template_input(namespace):
+    if namespace.template_file and not os.path.isfile(namespace.template_file):
+        raise CLIError('Please enter a valid file path')
+
+
+def _validate_template_spec(namespace):
+    if namespace.template_spec is None:
+        if (namespace.name is None or namespace.resource_group_name is None):
+            raise CLIError('incorrect usage: Please enter'
+                           'a resource group and resource name or a --template-spec resource ID')
+    else:
+        from msrestazure.tools import is_valid_resource_id
+        if not is_valid_resource_id(namespace.template_spec):
+            raise CLIError('--template-spec is not a valid resource ID.')
+
+
+def _validate_template_spec_out(namespace):
+    _validate_template_spec(namespace)
+    if namespace.output_folder and not os.path.isdir(namespace.output_folder):
+        raise CLIError('Please enter a valid output folder')
+
+
+def _validate_deployment_name_with_template_specs(namespace):
+    # If missing,try come out with a name associated with the template name
+    if namespace.deployment_name is None:
+        template_filename = None
+        if namespace.template_file and os.path.isfile(namespace.template_file):
+            template_filename = namespace.template_file
+        if namespace.template_uri and urlparse(namespace.template_uri).scheme:
+            template_filename = urlsplit(namespace.template_uri).path
+        if namespace.template_spec:
+            from msrestazure.tools import parse_resource_id, is_valid_resource_id
+            if not is_valid_resource_id(namespace.template_spec):
+                raise CLIError('--template-spec is not a valid resource ID.')
+            template_filename = parse_resource_id(namespace.template_spec).get('resource_name')
+        if template_filename:
+            template_filename = os.path.basename(template_filename)
+            namespace.deployment_name = os.path.splitext(template_filename)[0]
+        else:
+            namespace.deployment_name = 'deployment1'
```

Comment:
Does repeated conflicts occur if the name of deployment1 appears multiple times? Could you change it to a random value, for example, suffix it with UUID or timestamp?

Reply:
Doing so (suffix it) causes errors in other modules.

Reply:
What errors does it cause to other modules?

Reply:
Other modules have test that check the deployment name is equal to 'deployment1', adding a time stamp would cause these tests to fail